docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,193 @@
+
+[id="prefetching-dependencies-to-support-hermetic-build_{context}"]
+= Prefetching the package manager dependencies for the Hermetic build
+
+{ProductName} uses Cachi2 to prefetch the package manager dependencies for the <<supported-languages>>. Cachi2 is a wrapper around the package managers that makes it easy to prefetch and cache dependencies. It also helps generate a manifest of all dependencies included in your builds, ensuring transparency and maintainability.
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a `go.mod` file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: gomod <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `gomod`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip`
+Cachi2 supports pip by parsing of `pip` requirements files, including but not limited to *requirements.txt* files placed in the root of your repository. By generically parsing `pip` requirements files, it downloads the specified dependencies.
+
+IMPORTANT: The requirements file can have a different name, as you can use multiple files to provide the dependencies. These requirements files function as lockfiles, encompassing all the transitive dependencies. It's crucial to actively pin each transitive dependency listed in the requirements file to a specific version.
+
+
+s, which include all the transitive dependencies. Each transitive dependency in the *requirements.txt* file must be pinned to a specific version.
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a <filename> file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component’s source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: pip <1>
```

Comment:
What should they be modifying in their definition? The params section? or the default section? I have three references per pipeline when searching for 'prefetch-input', one is empty, two are variable references.
```
    - default: ""
      description: Build dependencies to be prefetched by Cachi2
      name: prefetch-input
      type: string
```
```
       - name: prefetch-dependencies
      params:
      - name: input
        value: $(params.prefetch-input)
```
        
```
        - name: PREFETCH_INPUT
        value: $(params.prefetch-input)
```

Reply:
They should be adding it to [.spec.params](https://github.com/containerbuildsystem/cachi2/blob/9917e6f3169bf74e661da1501417b8048a162204/.tekton/push.yaml#L12) I would say

```yaml
spec:
  params:
    - ...
    - name: prefetch-input
      value: ...
```

Reply:
It needs to be added so you will not be able to search it to find it. Therefore, it might be helpful to direct the user to exactly the place to make the addition.

Reply:
Shall I file a story for build-service to add all available params to the generated PipelineRuns in the user's .tekton dir?

Reply:
Hmm, since the entire Pipeline is inlined in the PipelineRun, I guess it doesn't really matter

Normally, the params in a Pipeline declare the available params while those in a PipelineRun set the values for that particular "run" of a Pipeline

```python
# the Pipeline
def my_pipeline(hermetic: bool = False):
   ...

# the PipelineRun
my_pipeline(hermetic=True)
```
But this would matter more if the pipeline was only referenced (as it is [in our cachi2 repo](https://github.com/containerbuildsystem/cachi2/blob/main/.tekton/pull-request.yaml) - we probably created the PipelineRun manually). Since the entire pipeline definition is there, users might as well change the defaults